longevity_tombstone_gc_test.py

```diff
@@ -0,0 +1,125 @@
+        wait for load to end + TTL period.
+        Run a major compaction.
+        Verify no tombstones.
+        """
+        # pylint: disable=too-many-locals
+
+        self.create_tables_for_scylla_bench()
+        self.db_node = self.db_cluster.nodes[0]
+        self.run_post_prepare_cql_cmds()
+        stress_queue = []
+
+        stress_cmd = self.params.get('stress_cmd')
+        params = {'stress_cmd': stress_cmd, 'round_robin': self.params.get('round_robin')}
+        self._run_all_stress_cmds(stress_queue, params)
+
+        self.log.info('Wait a duration of TTL + propagation_delay_in_seconds')
+        wait_for_tombstones = 4 * 60 * 2
+        time.sleep(wait_for_tombstones)
+        self.db_node.run_nodetool(f"flush -- {self.keyspace}")
+        sstable_utils = SstableUtils(db_node=self.db_node, propagation_delay_in_seconds=self.propagation_delay,
+                                     ks_cf=self.ks_cf)
+        self.log.info('Count the initial number of tombstones')
+        tombstone_num_pre_repair = sstable_utils.count_tombstones()
+        self._run_repair_and_major_compaction()
+        tombstone_num_post_repair = sstable_utils.count_tombstones()
+        assert tombstone_num_post_repair >= tombstone_num_pre_repair, \
+            f"Found unexpected fewer tombstones: {tombstone_num_post_repair} / {tombstone_num_pre_repair}"
+
+        self.log.info("change gc-grace-seconds back to default of 10 days and tombstone-gc mode to 'repair'")
+        with self.db_cluster.cql_connection_patient(node=self.db_node) as session:
+            query = "ALTER TABLE scylla_bench.test with gc_grace_seconds = 36240 " \
+                    f"and tombstone_gc = {{'mode': 'repair', 'propagation_delay_in_seconds':'{self.propagation_delay}'}};"
+            session.execute(query)
+
+        self._run_repair_and_major_compaction(wait_propagation_delay=True)
+
+        self.log.info("verify no tombstones exist in post-repair-created sstables")
+
+        table_repair_date, delta_repair_date_minutes = sstable_utils.get_table_repair_date_and_delta_minutes()
+        sstables = sstable_utils.get_sstables(from_minutes_ago=delta_repair_date_minutes)
+        self.log.debug('Starting sstabledump to verify correctness of tombstones for %s sstables',
+                       len(sstables))
+        for sstable in sstables:
+            sstable_utils.verify_post_repair_sstable_tombstones(table_repair_date=table_repair_date, sstable=sstable)
+
+        self.log.info("Change tombstone-gc mode to 'immediate'")
+        with self.db_cluster.cql_connection_patient(node=self.db_node) as session:
+            query = "ALTER TABLE scylla_bench.test with tombstone_gc = {'mode': 'immediate', 'propagation_delay_in_seconds':'300'};"
+            session.execute(query)
+
+        self.log.info('Wait a duration of schema-agreement and propagation_delay_in_seconds')
+        self.db_cluster.wait_for_schema_agreement()
+        time.sleep(self.propagation_delay)
```

Comment:
here you wait 2 times... why?

Reply:
Mostly to be on the safe side - first wait for all nodes to be synced with schema change, then wait propagation so user data is updated as well.